lib/ansible/module_utils/basic.py

```diff
@@ -391,7 +391,7 @@ def _remove_values_conditions(value, no_log_strings, deferred_removals):
             if omit_me in stringy_value:
                 return 'VALUE_SPECIFIED_IN_NO_LOG_PARAMETER'
 
-    elif isinstance(value, datetime.datetime):
+    elif isinstance(value, datetime.datetime) or isinstance(value, datetime.date):
```

Comment:
```suggestion
    elif isinstance(value, (datetime.datetime, datetime.date)):
```

Reply:
Please add a changelog entry like https://github.com/ansible/ansible/blob/devel/changelogs/fragments/565_default_callback.yml and unit test for this change.